### PR TITLE
Improve l10n on DatePickerPanel component

### LINKED ISSUE #304: Snooze Tabs panel becomes unresponsive after clicking the "Pick a Date/Time" option's arrow
[Affected versions]:
- Nightly 55.0a1
- Snooze Tabs 1.0.17

[Affected Platforms]:
- All Windows
- All MAC OS
- All Linux

[Prerequisites]:
- Have a Firefox profile with the latest Snooze Tabs add-on version (1.0.17 -dev, built on 12/04/2017) loaded as a temporary add-on.

[Steps to reproduce]:
1. Open the browser with the profile from prerequisite and navigate to any website.
2. Click on the "SnoozedTabs" button in the browser toolbar.
3. Click on the "Pick a Date/Time" arrow.

[Expected result]:
- "Pick a Date/Time" panel slides over.

[Actual result]:
- The "Options" panel is greyed out and unresponsive. 

[Notes]:
- The panel returns to normal if the user clicks outside the panel and goes back again to the Snooze Tabs button.
- This is also reproducible after moving the "Snooze Tabs" button in the Hamburger Menu.
- This is not reproducible with the Test Pilot version.
- Please see below the screen recording of the issue:

snz

### LINKED ISSUE #305: Localization of DateTime picker strings
A few minutes ago a lot of strings (29) landed and were exposed in Pontoon. I say a lot, given that the add-on so far consists of 34 strings.

I have asked @lmorchard to revert that change, because there are a few issues that I would like to discuss before exposing them for localization.
https://github.com/bwinton/SnoozeTabs/commit/76dfaca05285b646fab0ba9ef5c301aa4cb82a8a

My understanding from the IRC discussion is that they're used to localize this component
https://github.com/react-component/calendar/tree/master/src/locale

The first problem is that we shouldn't ask localizers to provide datetime formats. We have Intl APIs in Firefox, we should try to leverage that as much as possible.

There's also a datetime picker in Firefox, not sure in which versions is available and if it's a suitable alternative.

If we end up having to expose these strings, we need much better localization comments to explain what's going on. I also expect to be broken all over the place, with localizers picking wrong values because they don't understand what they're asked to do (I've worked on Firefox OS, I know that will happen).

CC @zbraniecki since he's doing all the work on Intl in Gecko.

### PR DESCRIPTION
**Pull request by lmorchard**
- Patch Moment.format() to use Intl.DateTimeFormat() API for rc-calendar
  locale formatters in DatePickerPanel

- Upgrade rc-calendar component to 8.1.0

- Import additional locales supplied by new version of rc-calendar

- Drop rc-calendar strings that don't seem reachable from our use case
  of the component

- Add more detailed comments before each of the remaining new strings

- Small bugfix for storybook

- Small bugfix for initially null datetime on "Pick" time option (#304)

Fixes #305.
Fixes #304.

### CODE REVIEW

**Comment by lmorchard** [comment]
Practically speaking, we probably don't need the strings about decades & centuries either - because who's going to snooze a tab for 10 - 100 years from now? 

But, there's not an easy way to disable those panels in the component, so it might be nice to at least have the labels so folks know what they're getting into when they click them.

File: locales/en-US/snoozetabs.properties [hunk 1]
```diff
@@ -62,34 +62,21 @@ manageUndo=Undo
...
-calendar_clear=Clear
-calendar_month=Month
-calendar_year=Year
-calendar_timeSelect=Select time
-calendar_dateSelect=Select date
+# Tooltip label for the table header in date picker that can be clicked to select a month.
 calendar_monthSelect=Choose a month
+# Tooltip label for the table header in date picker that can be clicked to select a year.
 calendar_yearSelect=Choose a year
+# Tooltip label for the table header in date picker that can be clicked to select a decade.
```
Comment (flodolo) [inline locales/en-US/snoozetabs.properties]:
```
# Tooltip displayed in the date picker when hovering the control to select a decade.
# Click on the year in the header to access the decade/year selection.
```

File: locales/en-US/snoozetabs.properties [hunk 2]
```diff
@@ -62,34 +62,21 @@ manageUndo=Undo
...
 calendar_decadeSelect=Choose a decade
-calendar_yearFormat=YYYY
-calendar_dateFormat=M/D/YYYY
-calendar_dayFormat=D
-calendar_dateTimeFormat=M/D/YYYY HH:mm:ss
-calendar_monthFormat=MMMM
-# true if the month name should appear before the year when displayed, false if not
-calendar_monthBeforeYear=true
-# (PageUp) is a description of the keyboard shortcut
+# Tooltip label for the arrow button in date picker to move to the previous month. (PageUp) is a description of the keyboard shortcut
```
Comment (flodolo) [inline locales/en-US/snoozetabs.properties]: You can go on two lines for better readability.

```
# Tooltip for the arrow button in date picker to move to the previous month.
# (PageUp) is a description of the keyboard shortcut.
```

File: locales/en-US/snoozetabs.properties [hunk 3]
```diff
@@ -62,34 +62,21 @@ manageUndo=Undo
...
-calendar_backToToday=Back to today
-calendar_ok=Ok
-calendar_clear=Clear
-calendar_month=Month
-calendar_year=Year
-calendar_timeSelect=Select time
-calendar_dateSelect=Select date
+# Tooltip label for the table header in date picker that can be clicked to select a month.
 calendar_monthSelect=Choose a month
+# Tooltip label for the table header in date picker that can be clicked to select a year.
```
Comment (flodolo) [inline locales/en-US/snoozetabs.properties]:
```
# Tooltip label displayed in the date picker when hovering the current year in the header
```

File: locales/en-US/snoozetabs.properties [hunk 4]
```diff
@@ -62,34 +62,21 @@ manageUndo=Undo
...
-# true if the month name should appear before the year when displayed, false if not
-calendar_monthBeforeYear=true
-# (PageUp) is a description of the keyboard shortcut
+# Tooltip label for the arrow button in date picker to move to the previous month. (PageUp) is a description of the keyboard shortcut
 calendar_previousMonth=Previous month (PageUp)
-# (PageDown) is a description of the keyboard shortcut
+# Tooltip label for the arrow button in date picker to move to the next month. (PageDown) is a description of the keyboard shortcut
 calendar_nextMonth=Next month (PageDown)
-# (Control + left) is a description of the keyboard shortcut
-calendar_previousYear=Last year (Control + left)
```
Comment (flodolo) [inline locales/en-US/snoozetabs.properties]: We still need this (and its sibling), it's displayed when you click on the month name.

```
# Tooltip for the arrow to move to the previous year in month selection
```

File: locales/en-US/snoozetabs.properties [hunk 5]
```diff
@@ -62,34 +62,21 @@ manageUndo=Undo
...
-# (Control + right) is a description of the keyboard shortcut
-calendar_nextYear=Next year (Control + right)
+# Tooltip label for the arrow button to move to the previous decade in year selection
```
Comment (flodolo) [inline locales/en-US/snoozetabs.properties]:
```
# Tooltip for the arrow to move to the previous decade in year selection
```
(similar to the next command)

File: locales/en-US/snoozetabs.properties [hunk 6]
```diff
@@ -62,34 +62,21 @@ manageUndo=Undo
...
 calendar_previousDecade=Last decade
+# Tooltip label for the arrow button to move to the next decade in year selection
 calendar_nextDecade=Next decade
+# Tooltip label for the arrow button to move to the previous century in year selection
```
Comment (flodolo) [inline locales/en-US/snoozetabs.properties]: decade selection
```
# Tooltip for the arrow to move to the previous century in decade selection.
# To access the decade selection, click on the current decade header in year selection.
```

**Comment by flodolo** [comment]
Note sure if @zbraniecki wants to take a quick look at the code

**Comment by lmorchard** [comment]
Updated with revised comments in snoozetabs.properties. Dropped the code to try using the locales that come with rc-calendar.

File: locales/en-US/snoozetabs.properties [hunk 7]
```diff
@@ -62,34 +62,26 @@ manageUndo=Undo
...
-calendar_monthBeforeYear=true
-# (PageUp) is a description of the keyboard shortcut
+# Tooltip for the arrow button in date picker to move to the previous month.
+# (PageUp) is a description of the keyboard shortcut.
 calendar_previousMonth=Previous month (PageUp)
+# Tooltip label for the arrow button in date picker to move to the next month.
 # (PageDown) is a description of the keyboard shortcut
 calendar_nextMonth=Next month (PageDown)
-# (Control + left) is a description of the keyboard shortcut
-calendar_previousYear=Last year (Control + left)
```
Comment (flodolo) [inline locales/en-US/snoozetabs.properties]: You still need to resurrect these two strings

**Comment by flodolo** [comment]
There are two strings that are still needed, but it looks much better, thanks.
https://github.com/bwinton/SnoozeTabs/pull/306#discussion_r111318178

**Comment by lmorchard** [comment]
Updated to include those two strings

**Comment by lmorchard** [comment]
Going to merge to master. Does this look good enough to merge back into the l10n branch, now?

**Comment by flodolo** [comment]
> Going to merge to master. Just to confirm - does this look good enough to merge back into the l10n branch, now?

Yes, absolutely.

**Comment by lmorchard** [comment]
Awesome! Thanks for all the help!

**Pull request merged**